services/setup/public/index.html

```diff
@@ -1,26 +1,63 @@
 <!DOCTYPE html>
 <html>
-
 <head>
   <title>Radiodan Configuration</title>
   <style>
+    .hidden { display: none; }
     button, label, span { display: block }
     button { margin: 10px 0; width: 50% }
     span { float: left; width: 50%; }
   </style>
 </head>
-
 <body>
 <h1>Radiodan Configuration</h1>
+<p id="notification" class="hidden">
+  Changes have been saved, please reboot your Pi for them to take effect.
+</p>
+
+<h2>Application</h2>
+<form action="/apps" method="POST" id="apps-list">
+  <label>
+    <span>All Apps</span>
+    <input type="radio" name="apps" value="" checked="checked" />
+  </label>
+  <button type="submit">Update</button>
+</form>
 
 <h2>Services</h2>
 <form action="/services" method="POST" id="services-list">
   <button type="submit">Update</button>
 </form>
 
 <script>
-  const app = async () => {
+  const buildApps = async () => {
+    const form = document.getElementById('apps-list');
+    const selection = document.createElement('div');
+    const apps = await fetch('/apps').then(res => res.json());
+
+    apps.forEach(({ name, active }) => {
+      const label = document.createElement('label');
+      const span = document.createElement('span');
```

Comment:
You could also do something like:

```js
label.innerHTML = `
  <span>${name}</span>
  <input type="radio" name="apps" value=${name} ${active ? "checked" : ""}  />
`;
selection.appendChild(label);
```

As long as you don't need a reference to the `span` or `radio` elements here.